### ISSUE TITLE
Excessive output if git is not installed

### ISSUE THREAD

**Issue by ndmitchell** [commercialhaskell/stack#2521]
On a brand new machine (64bit Windows), I downloaded the latest 1.1.2 as a binary and put it on the PATH, ran `stack setup`, and faced a wall of:

```
Left C:\Users\Neil\AppData\Roaming\stack\indices\Hackage\git-update\all-cabal-hashes\.git\objects\pack: getDirectoryContents: does not exist (The system cannot find the path specified.)
Did not find .cabal file for Agda-2.5.1 with Git SHA of cb3ed7a8d5df3cd50b497637fc317f020d1bf59d
Left C:\Users\Neil\AppData\Roaming\stack\indices\Hackage\git-update\all-cabal-hashes\.git\objects\pack: getDirectoryContents: does not exist (The system cannot find the path specified.)
Did not find .cabal file for AC-Vector-2.3.2 with Git SHA of 32d97a60969fe6912b33597e7399c407b5f2e39a
Left C:\Users\Neil\AppData\Roaming\stack\indices\Hackage\git-update\all-cabal-hashes\.git\objects\pack: getDirectoryContents: does not exist (The system cannot find the path specified.)
Preparing to install GHC to an isolated location.
```

This went on for 100's of lines, perhaps even 1000's. Is the `Left` debug droppings that got left? A problem? Not a problem? It hasn't yet lead to a problem, as far as I can tell... But each install does print out a bunch of:

```
Did not find .cabal file for unordered-containers-0.2.7.1 with Git SHA of 458a2a10936d8e79d96b021505120d1598311fbe
Left C:\Users\Neil\AppData\Roaming\stack\indices\Hackage\git-update\all-cabal-hashes\.git\objects\pack: getDirectoryContents: does not exist (The system cannot find the path specified.)
```

**Comment by ndmitchell** [comment]
Installing git and putting it on the $PATH made the large number of warnings go away. This translates to a poor experience for those without git.

**Comment by mgsloan** [comment]
Dupe of https://github.com/commercialhaskell/stack/issues/2241 , which is resolved on master.  It's about to be released as v0.1.2
